src/smif/data_layer/data_handle.py

```diff
@@ -291,38 +323,66 @@ def get_data(self, input_name: str, timestep=None) -> DataArray:
                                                         self._timesteps)
             else:
                 assert isinstance(timestep, int) and timestep <= self._current_timestep
+        return timestep
 
-        # resolve source
-        dep = self._resolve_source(input_name)
-
-        source_model_name = dep['source_model_name']
-        source_output_name = dep['source_output_name']
-
-        self.logger.debug(
-            "Read %s %s %s", source_model_name, source_output_name, timestep)
-
-        spec = self._inputs[input_name]
-
-        if dep['type'] == 'scenario':
-            data = self._store.read_scenario_variant_data(
-                source_model_name,  # read from a given scenario model
-                dep['variant'],  # with given scenario variant
-                source_output_name,  # using output (variable) name
-                timestep
-            )
-        else:
+    def get_result(self, dep, timestep, spec):
```

Comment:
Should this be private? Similarly for `get_scenario` below - if we want to encourage use of the generic `get_data` methods.